### PR TITLE
Extract NodeController from l7 lb

### PR DESCRIPTION
**Pull request by panslava**

### CODE REVIEW

File: pkg/controller/node.go [hunk 1]
```diff
@@ -70,13 +73,15 @@ func NewNodeController(ctx *context.ControllerContext) *NodeController {
 // Run the queue to process updates for the controller. This must be run in a
 // separate goroutine (method will block until queue shutdown).
 func (c *NodeController) Run() {
+	defer c.Shutdown()
```
Comment (kl52752) [inline pkg/controller/node.go:76]: this will not work this queue is blocking (see comment above :))

Comment (kl52752) [reply pkg/controller/node.go:76]: fixed

File: pkg/controller/node.go [hunk 2]
```diff
@@ -70,13 +73,15 @@ func NewNodeController(ctx *context.ControllerContext) *NodeController {
...
 	start := time.Now()
 	for !c.hasSynced() {
 		klog.V(2).Infof("Waiting for hasSynced (%s elapsed)", time.Now().Sub(start))
 		time.Sleep(1 * time.Second)
 	}
 	klog.V(2).Infof("Caches synced (took %s)", time.Now().Sub(start))
-	c.queue.Run()
+	go c.queue.Run()
+	<-c.stopCh
```
Comment (prameshj) [inline pkg/controller/node.go:84]: Maybe add c.Shutdown after this line, instead of defer?

Comment (panslava) [reply pkg/controller/node.go:83]: Agree, this feels safer, fixed!

**Pull request merged**